---
output:
  md_document:
    variant: markdown_github
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "README-"
)

library(psiplot)
```

[![Build Status](https://travis-ci.org/kcha/psiplot.svg?branch=master)](https://travis-ci.org/kcha/psiplot)

# psiplot

psiplot is an R package for generating plots of percent spliced-in (PSI) values of 
alternatively-spliced exons that were computed by 
[VAST-TOOLS](https://github.com/vastgroup/vast-tools), an RNA-Seq pipeline for 
alternative splicing analysis. The plots are generated using `ggplot2`.


For a demo of psiplot, take a look at the companion Shiny app here: http://kcha.shinyapps.io/psiplotter-app.

## Installation

See [Releases](https://github.com/kcha/psiplot/releases) for the latest release
or get the most up-to-date development version via devtools:
```{r eval=FALSE}
install.packages("devtools")
devtools::install_github("kcha/psiplot")
```

## Usage

### Input
psiplot takes as input the PSI and/or cRPKM results generated by VAST-TOOLS 
(e.g. after running `vast-tools combine` or `vast-tools diff`). This README
uses the provided sample datasets, `psi` and `crpkm`, as example input data.

### Plotting 
The function `plot_event()` generates a plot of a single event:

```{r eval=FALSE}
library(psiplot)

# Plot an event using provided sample dataset
plot_event(psi[1,])

# Alternatively, to plot an event by gene name (for example, TSPAN6):
plot_event(psi[psi$GENE == "TSPAN6",])
```

In addition, cRPKM expression values generated by VAST-TOOLS can also be plotted
in a similar manner using the function `plot_expr()`:
```{r eval=FALSE}
# Using sample dataset
plot_expr(crpkm[1,])

# Alternatively
plot_expr(crpkm[crpkm$NAME == "TSPAN6",])
```

### Customizing plots

There are two ways to customize the plots: using a configuration file or using R
arguments.

#### The `.config` file way
In `vast-tools plot`, an optional config file can be used to customize the
plots' visual appearance. The same config file can be supplied here as well.

```{r eval=FALSE}
plot_event(psi[1,], config="/path/to/config")

# config can also be pre-loaded into a data frame
cfg <- read.table("/path/to/config", header=T, sep="\t", stringsAsFactor=FALSE)
plot_event(psi[1,], config=cfg)
```

The color and ordering of samples can be customized by supplying a plot
configuration file. This file is tab-delimited and must be
manually created. For example:

```{r}
# sample config data
config
```

- **Order**: The ordering of the samples from left to right.
- **SampleName**: Name of the sample. MUST match sample name in input table.
- **GroupName**: Group name. Use for plotting the average PSI of samples belonging
to the same group (enable by setting `groupmean=TRUE`)
- **RColorCode**: An R color specification:
    1. color name (as specified by `colors()`)
    2. hex color code (#rrggbb)

Tips:

- The samples under SampleName MUST MATCH the names in the input table.
Only the samples listed in the config file will be represented in the
resulting plots. Other samples in the input table but not in the config
file will be ignored. This may be useful for plotting only a subset of samples.
- The column **Order** does not need to be in sorted order. Thus, you can change
the ordering of your samples by simply changing the order number.

#### The R way

The colors and other graphical parameters can also be configured in R via arguments.
`plot_events()` and `plot_expr()` provides a limited set of arguments and can be 
used in conjunction with a config file. See `?plot_events` and `?plot_expr` for 
more details on the available options. 

```{r eval=FALSE}
plot_event(psi[1,], config = config, pch = 9, ylim = c(20, 80))
```

## Issues
Please report all bugs and issues using the [issue
tracker](https://github.com/kcha/psiplot/issues).

## Related Projects
- [VAST-TOOLS](https://github.com/vastgroup/vast-tools)
- [psiplotter-app](https://github.com/kcha/psiplotter-app): A Shiny app for visualizing PSI plots

## Acknowledgements
- Manuel Irimia
- Nuno Barbosa-Morais
- Tim Sterne-Weiler
